haystack/dataclasses/document.py

```diff
@@ -115,16 +112,12 @@ def __post_init__(self):
         # Generate an id only if not explicitly set
         self.id = self.id or self._create_id()
 
-        if self.dataframe is not None:
-            msg = "The `dataframe` field is deprecated and will be removed in Haystack 2.11.0."
-            warnings.warn(msg, DeprecationWarning)
-
     def _create_id(self):
         """
         Creates a hash of the given content that acts as the document's ID.
         """
         text = self.content or None
-        dataframe = self.dataframe.to_json() if self.dataframe is not None else None
+        dataframe = None  # this allows the ID creation to remain unchanged even if the dataframe field has been removed
```

Comment:
This ensures that all IDs for Documents not containing a dataframe will be unchanged after this PR.

If we drop this idea, this would mean:
- introducing a subtle change in ID generation, which we should communicate to users
- changing several tests (not a big issue)

LMK what you think

Reply:
Yeah I think fine to leave in